### PR TITLE
Add new version of `--summary` option to `dolt diff`

### PR DESCRIPTION
**Pull request by tbantle22**
The old version of `dolt diff --summary` was renamed to `dolt diff --stat`
This new version of `dolt diff --summary` returns a summary of what tables changed/how between two refs
```
us-jails % dolt diff --summary
+-----------------------------+-----------+-------------+---------------+
| Table name                  | Diff type | Data change | Schema change |
+-----------------------------+-----------+-------------+---------------+
| inmate_population_snapshots | dropped   | true        | true          |
| jails                       | modified  | true        | false         |
| test                        | added     | false       | true          |
+-----------------------------+-----------+-------------+---------------+
```

### CODE REVIEW

**Comment by tbantle22** [comment]
Go tests are failing from gms bump that was merged with errors

File: go/cmd/dolt/commands/diff.go [hunk 1]
```diff
@@ -467,6 +479,41 @@ func maybeResolve(ctx context.Context, dEnv *env.DoltEnv, spec string) (*doltdb.
 	return root, true
 }
 
+var diffSummarySchema = sql.Schema{
+	&sql.Column{Name: "Table name", Type: types.Text, Nullable: false},
+	&sql.Column{Name: "Diff Type", Type: types.Text, Nullable: false},
```
Comment (fulghum) [inline go/cmd/dolt/commands/diff.go:484]: (minor/nit) `Diff type` to be consistent with the other columns' capitalization?

```suggestion
	&sql.Column{Name: "Diff type", Type: types.Text, Nullable: false},
```

File: go/libraries/doltcore/sqle/dtables/unscoped_diff_table.go [hunk 2]
```diff
@@ -241,20 +233,20 @@ func (d *doltDiffWorkingSetRowItr) Next(ctx *sql.Context) (sql.Row, error) {
 		return nil, io.EOF
 	}
 
-	change, err := processTableDelta(ctx, tableDelta)
+	change, err := tableDelta.GetSummary(ctx)
```
Comment (fulghum) [inline go/libraries/doltcore/sqle/dtables/unscoped_diff_table.go:236]: Awesome! 🙌  Nice work cleaning this up!!

**Review by fulghum — APPROVED** [review]
Looks great! That refactoring for `TableDelta` turned out great. 🚢

**Pull request merged**